atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -580,6 +584,7 @@ private void extractLocalWritesForRow(@Output Map<Cell, byte[]> result,
         }
 
         Map<Cell, byte[]> result = getFromKeyValueService(tableRef, cells);
+        validateExternalAndCommitLocksIfNecessary(tableRef);
 
         return Maps.filterValues(result, Predicates.not(Value.IS_EMPTY));
```

Comment:
Guessing this is fine, because we do an identical sequence of things in the above `get` method.